src/OmniSharp.Script/FileParser.cs

```diff
@@ -0,0 +1,60 @@
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace OmniSharp.Script
+{
+    public class FileParser
+    {
+        public static FileParserResult ProcessFile(string path)
+        {
+            var result = new FileParserResult();
+            ParseFile(path, result);
+            return result;
+        }
+
+        private static void ParseFile(string path, FileParserResult result)
+        {
+            var fullPath = Path.GetFullPath(path);
+            if (result.LoadedScripts.Contains(fullPath))
+            {
+                return;
+            }
+
+            result.LoadedScripts.Add(fullPath);
+
+            var scriptCode = File.ReadAllText(fullPath);
+
+            var syntaxTree = CSharpSyntaxTree.ParseText(scriptCode, CSharpParseOptions.Default.
+                WithPreprocessorSymbols("load", "r").
```

Comment:
Is this line necessary? This would have the effect of implicitly adding the following code at that top of the tree:

``` C#
#define load
#define r
```

Is that what you meant?